web/resource-proxy/alloy-light.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+# Given a URL from the proxy config and an ID, look up the unit type
+
+class XmlParser {
+    public $results = array();
+
+    function __construct($f = "proxy.config") {
+    $file = dirname(__FILE__) . "/../../conf/$f";
+    $xml = file_get_contents($file);
+        $parser = xml_parser_create();
+        xml_set_object($parser, $this);
+        xml_set_element_handler($parser, "tagStart", "tagEnd");
+        xml_parse($parser, $xml);
+        xml_parser_free($parser);
+    }
+
+    function tagStart($parser, $name, $attrs) {
+        $attrs = array_change_key_case($attrs, CASE_LOWER);
+        $tag = array(strtolower($name) => $attrs);
+        array_push($this->results, $tag);
+    }
+
+    function tagEnd($parser, $name) {
+        $this->results[count($this->results)-2]['childrens'][] = $this->results[count($this->results)-1];
+        array_pop($this->results);
+    }
+}
+
+header("Content-Type: application/json");
+
+$id = isset($_GET['id']) ? $_GET['id'] : null;
```

Comment:
I think this should be validated and rejected if it doesn't match `^[a-f0-9]{24}$`, just as a precaution.